src/transformation/visitors/call.ts

```diff
@@ -290,12 +293,44 @@ export const transformCallExpression: FunctionVisitor<ts.CallExpression> = (node
     return wrapResultInTable ? wrapInTable(callExpression) : callExpression;
 };
 
-function isContextualCallExpression(context: TransformationContext, signature: ts.Signature | undefined): boolean {
-    const declaration = signature?.getDeclaration();
-    if (!declaration) {
-        return !context.options.noImplicitSelf;
+function isContextualCallExpression(
+    context: TransformationContext,
+    calledExpression: ts.Expression,
+    signature?: ts.Signature
+): boolean {
+    const symbol = context.checker.getSymbolAtLocation(calledExpression);
+    let hasNoSelfInFile = false;
+
+    if (symbol?.declarations) {
+        for (const declaration of symbol.declarations) {
+            const srcFile = declaration.getSourceFile();
+
+            hasNoSelfInFile = getFileAnnotations(srcFile).has(AnnotationKind.NoSelfInFile);
+            break;
+        }
```

Comment:
This is just not right, you only check the first declaration and break, wether it had noselfinfile or not. You only want to break when the annotation was found. I would just write this entire thing as
```ts
const hasNoSelfInFile = symbol?.declarations.some(d => getFileAnnotations(d.getSourceFile()).has(AnnotationKind.NoSelfInFile)) ?? false;
```

Reply:
Ah, I was assuming that there would anyways only be one file that it is declared in, my fault then.

Reply:
My latest commit should resolve what was noted here.